### DISCUSSION TITLE
generateMetadata doesn't work with static export?

### DISCUSSION THREAD

**Discussion by AndresSalomon1990** [vercel/next.js#55476] [App Router] [answered]
Hello, I'm building a simple webapp where I show a list of products and then go to the product page. There I grab the product name from a query param and render the product page based on that value (grab data and images from an asset folder).

Also, in that product page, I'm exporting the generateMetadata function to create a title based on that query param. Doing this for Google Analytics 4 so it tracks the title of the page that is visited.

`export function generateMetadata({ searchParams }: Props): Metadata {
  const { product } = searchParams;

  return {
    title: `${product}`,
  };
}`

The problem is that my client has a server that needs the static export, so I add the config `output: 'export'` to next.config, generate the out folder and upload that content to the server folder with Filezilla.

Everything works great but when going to the product page I'm getting `undefined` for the title Metadata prop.

What I'm missing? Is because this generateMetadata function is not supported for static exports? If so, how can I achive a similar result?

Thanks!!

**Comment by lr0pb** [comment, marked as answer]
Static export does not support runtime dynamic functions like `searchParams`, however it will work fine with dynamic route `[product]` that have `generateStaticParams`
Like this:
```ts
export function generateMetadata({ params }: Props): Metadata {
  const { product } = params;
  ...
}
```

**Reply by AndresSalomon1990** [reply]
Nice! I understand it now, I was missing `generateStaticParams`. I mean, static export needs `generateStaticParams`. Thanks for your answer :)

**Reply by landed1** [reply]
For some reason I am not seeing the meta getting set while using generateStaticParams. Is it possible that latency is the issue? And can I see the meta tags working in dev server? When I check the params in the meta function it is only { slug: 'extra' }
